src/plugins/xabsl/XABSL-Soccer.patch

```diff
@@ -0,0 +1,1009 @@
+ 			#if tag.associative
+ 			case tag
+-			when OrTag, AndTag:
++			when OrTag, AndTag
+ 				[0,3].each {|i|
+ 					child = val[i]
+ 					if(tag.class == child.class)
+@@ -1105,37 +1105,37 @@
+ 			tag = nil
+ 			
+ 			case symbol #the tag that defined the symbol name
+-			when DecimalInputSymbol:
++			when DecimalInputSymbol
+ 				tag = DISRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+-			when BooleanInputSymbol:
++			when BooleanInputSymbol
+ 				tag = BISRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+-			when EnumInputSymbol:
++			when EnumInputSymbol
+ 				tag = EISRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+ 				tag.enum_domain = symbol.enum_domain
+-			when DecimalOutputSymbol:
++			when DecimalOutputSymbol
+ 				tag = DOSRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+-			when BooleanOutputSymbol:
++			when BooleanOutputSymbol
+ 				tag = BOSRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+-			when EnumOutputSymbol:
++			when EnumOutputSymbol
+ 				tag = EOSRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+ 				tag.enum_domain = symbol.enum_domain
+-			when ConstantSymbol:
++			when ConstantSymbol
+ 				tag = CRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+-			when OptionParameterSymbol:
++			when OptionParameterSymbol
+ 				tag = OPRTag.new
+ 				tag.attr[ATTR_REF] = symbol
+ 				tag.out_type = symbol.type
+ 				if symbol.type == :enum
+ 					tag.enum_domain = symbol.enum_domain
+ 				end
+-			when EnumSymbolValueSymbol:
++			when EnumSymbolValueSymbol
+ 				tag = EVTag.new
+ 				tag.attr[ATTR_REF] = symbol
+ 			  tag.enum_domain = symbol.enum_domain
+@@ -1156,11 +1156,11 @@
+ 		def function val
+ 		  pc.enum_domains.pop
+ 		  case pc.function_dpcs.last.domain_symbol
+-		  when DecimalInputSymbol:
++		  when DecimalInputSymbol
+ 			  tag = DISRTag.new
+-			when BooleanInputSymbol:
++			when BooleanInputSymbol
+ 			  tag = BISRTag.new
+-			when EnumInputSymbol:
++			when EnumInputSymbol
+ 				tag = EISRTag.new
+ 			  tag.enum_domain = pc.function_dpcs.last.domain_symbol.enum_domain
+ 			else
+diff -urN XABSL/Xabsl/compiler/tokenizer.rb my-XABSL/Xabsl/compiler/tokenizer.rb
+--- XABSL/Xabsl/compiler/tokenizer.rb	2006-09-26 18:39:55.000000000 +0200
++++ my-XABSL/Xabsl/compiler/tokenizer.rb	2018-05-29 23:33:38.991683804 +0200
+@@ -182,9 +182,9 @@
+ 					when /\A((.|\n|\r)*?)(\*\/)/
+ 						
+ 						case comment_mode
+-						when :doc_comment:
++						when :doc_comment
+ 							push(:DOC_COMMENT, $1.strip)
+-						when :initial_comment:
++						when :initial_comment
+ 							push(:INITIAL_COMMENT, $1.strip)
+ 						end
+ 						str = $'
+diff -urN XABSL/Xabsl/compiler/xabsl.rb my-XABSL/Xabsl/compiler/xabsl.rb
+--- XABSL/Xabsl/compiler/xabsl.rb	2006-09-26 18:39:55.000000000 +0200
++++ my-XABSL/Xabsl/compiler/xabsl.rb	2018-05-29 23:31:38.696749724 +0200
+@@ -9,7 +9,7 @@
+ require 'translator'
+ require 'xabsl_parser.tab'
+ require 'fileutils'
+-require 'win32/registry'
++#require 'win32/registry'
+ require 'vcproj'
+ require 'ftools'

```

Comment:
```
Traceback (most recent call last):
	2: from /home/thofmann/code/fawkes/src/plugins/xabsl/XABSL/Xabsl/compiler/xabsl.rb:14:in `<main>'
	1: from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require'
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require': cannot load such file -- ftools (LoadError)
```

According to [this SO post](https://stackoverflow.com/questions/11552524/ruby-cannot-load-such-file-ftools-load-error), `ftools` should be replaced by `fileutils`:
```suggestion
-require 'ftools'
+require 'fileutils'
```